File: ql_lich_trinh_giang_day/app/Models/LopHocPhan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class LopHocPhan extends Model
{
    protected $table = 'LopHocPhan';
    protected $primaryKey = 'MaLHP';
    public $timestamps = false;
    protected $fillable = [
        'TenLHP',
        'MaMonHoc',
        'MaHK',
        'MaPhongMacDinh',
        'SiSoToiDa'
    ];
    protected $attributes = [
        'SiSoToiDa' => 50,
    ];


    public function monHoc()
    {
        return $this->belongsTo(MonHoc::class, 'MaMonHoc', 'MaMonHoc');
    }

    public function hocKy()
    {
        return $this->belongsTo(HocKy::class, 'MaHK', 'MaHK');
    }

    public function phongMacDinh()
    {
        return $this->belongsTo(PhongHoc::class, 'MaPhongMacDinh', 'MaPhong');
    }

    public function phanCongs()
    {
        return $this->hasMany(PhanCong::class, 'MaLHP', 'MaLHP');
    }

    public function giangViens()
    {
        return $this->belongsToMany(GiangVien::class, 'PhanCong', 'MaLHP', 'MaGV')
            ->withPivot('NgayPhanCong');
    }

    public function sinhViens()
    {
        return $this->belongsToMany(SinhVien::class, 'DangKyHocPhan', 'MaLHP', 'MaSV')
            ->withPivot('NgayDangKy');
    }

    public function lichTrinhChiTiet()
    {
        return $this->hasMany(LichTrinhChiTiet::class, 'MaLHP', 'MaLHP');
    }
}

File: ql_lich_giang_day_app/lib/models/lophocphan.dart
class LopHocPhan {
  final int maLHP;
  final String tenLHP;
  final int maMonHoc;
  final int maHK;
  final int? maPhongMacDinh;
  final int siSoToiDa;
  final String trangThai;

  LopHocPhan({
    required this.maLHP,
    required this.tenLHP,
    required this.maMonHoc,
    required this.maHK,
    this.maPhongMacDinh,
    required this.siSoToiDa,
    required this.trangThai,
  });

  factory LopHocPhan.fromJson(Map<String, dynamic> json) => LopHocPhan(
    maLHP: json['MaLHP'],
    tenLHP: json['TenLHP'],
    maMonHoc: json['MaMonHoc'],
    maHK: json['MaHK'],
    maPhongMacDinh: json['MaPhongMacDinh'],
    siSoToiDa: json['SiSoToiDa'],
    trangThai: json['TrangThai'],
  );

  Map<String, dynamic> toJson() => {
    'MaLHP': maLHP,
    'TenLHP': tenLHP,
    'MaMonHoc': maMonHoc,
    'MaHK': maHK,
    'MaPhongMacDinh': maPhongMacDinh,
    'SiSoToiDa': siSoToiDa,
    'TrangThai': trangThai,
  };
}
